CMakeLists.txt

```diff
@@ -161,6 +167,28 @@ else()
 
 endif()
 
+if(BUILD_UVWASI_SUPPORT)
+  if(BUILD_WASI_SUPPORT)
+    message(FATAL_ERROR "BUILD_UVWASI_SUPPORT is incompatible with BUILD_WASI_SUPPORT")
+  endif()
+
+  include(FetchContent)
+  FetchContent_Declare(
+    uvwasi
+    GIT_REPOSITORY https://github.com/cjihrig/uvwasi.git
+    GIT_TAG ${v0.0.7}
+  )
+
+  FetchContent_GetProperties(uvwasi)
+  if(NOT uvwasi_POPULATED)
+    FetchContent_Populate(uvwasi)
+    include_directories("${uvwasi_SOURCE_DIR}/include")
```

Comment:
The build fails on my Ubuntu machine with this error:
```log
In file included from ../source/m3_api_uvwasi.c:25:0:
_deps/uvwasi-src/include/uvwasi.h:8:10: fatal error: uv.h: No such file or directory
 #include "uv.h"
          ^~~~~~
```
I resolved it by adding `include_directories("${libuv_SOURCE_DIR}/include")` here.
Not sure if it's a correct way to fix it.

Reply:
Same as https://github.com/nodejs/wasi/issues/26

Reply:
That seems right. I think it probably worked on my machine because I have libuv installed on the system already. Sorry about that.